sim/core/spell_mod.go

```diff
@@ -65,18 +69,23 @@ func buildMod(unit *Unit, config SpellModConfig) *SpellMod {
 		removeFn = functions.Remove
 	}
 
+	if config.ResourceType > 0 && (config.ResourceType&proto.ResourceType_ResourceTypeMana|proto.ResourceType_ResourceTypeEnergy|proto.ResourceType_ResourceTypeRage|proto.ResourceType_ResourceTypeFocus == 0) {
+		panic("ResourceType " + strconv.Itoa(int(config.ResourceType)) + " for SpellMod is not implemented")
+	}
```

Comment:
I think a simple `%s` on `config.ResourceType` should work for a descriptive error message, since `ResourceType` implements `String()`.